test/files/pjs003/fail-pjs003-IFC4_3_ADD2_GuidTests.ifc

```diff
@@ -0,0 +1,42 @@
+ISO-10303-21;
+HEADER;
+
+FILE_DESCRIPTION((''),'2;1');
+FILE_NAME('','2024-06-12T09:50:55',(''),(''),'LAB','GMN3DS','');
+FILE_SCHEMA(('IFC4X3_ADD2'));
+ENDSEC;
+
+DATA;
+
+/* PROJECT CONTEXT  */
+#1= IFCPROJECT('2eS$fZmw9QmetNIywDEzSR',#2,'ODT',$,$,$,$,(#10),#15);
+#2= IFCOWNERHISTORY(#3,#6,$,.NOCHANGE.,$,$,$,152828);
+#3= IFCPERSONANDORGANIZATION(#4,#5,$);
+#4= IFCPERSON($,'','ALCHEMIST',$,$,$,$,$);
+#5= IFCORGANIZATION($,'LAB',$,$,$);
+#6= IFCAPPLICATION(#5,'2024','GREY','REVOLT');
+
+#10= IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,1.0E-05,#11,$);
+#11= IFCAXIS2PLACEMENT3D(#12,$,$);
+#12= IFCCARTESIANPOINT((0.,0.,0.));
+
+#15= IFCUNITASSIGNMENT((#16));
+#16= IFCSIUNIT(*,.LENGTHUNIT.,.MILLI.,.METRE.);
+
+
+/* SITES AGGREGATED TO PROJECT */
+#29=IFCRELAGGREGATES('2te7nJfPvJyvPEahDDrZW0',#2,'ProjectContainer',$,#1,(#29990,#30000,#30010,#30020,#30030,#30040));
+#31=IFCLOCALPLACEMENT($,#11);
+
+ 
+#29990=IFCSITE('14oxcl5LHMZQ6iCWJKpe5w',#2,'SITE_withCorrectGuid',$,$,#31,$,$,.COMPLEX.,$,$,$,$,$);
+#30000=IFCSITE(30000,#2,'SITE_withIncorrectGuid','GUID is not a string',$,#31,$,$,.COMPLEX.,$,$,$,$,$); 

```

Comment:
```suggestion
#30000=IFCSITE('30000',#2,'SITE_withIncorrectGuid','GUID is not a string',$,#31,$,$,.COMPLEX.,$,$,$,$,$); 
```
`GlobalId` needs to be a string to avoid schema errors.

Reply:
True, this was the testfile that I initially used. Two of them were indeed triggered schema-errors already indeed, three were to be implemented by us.[ I've deleted the two lines](https://github.com/buildingSMART/ifc-gherkin-rules/pull/289/commits/c79b759ffe4363be3200e12a65b9b1e2830b2c7d)
Furthermore, the validation stopped when `inst` had an invalid type. We should probably implement something that prevents the validation from silently stopping for the rest of the instances in the sequence when one of them causes an error within the step implementation. For now, I've implemented a[ solution for the current step_imp](https://github.com/buildingSMART/ifc-gherkin-rules/pull/289/commits/c79b759ffe4363be3200e12a65b9b1e2830b2c7d).